### PR TITLE
Fixed long overflow in some places. Version bump

### PR DESCRIPTION
**Pull request by 0486**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/cc/CCBidOrderPlacementTransactionType.java [hunk 1]
```diff
@@ -61,11 +61,18 @@ public ColoredCoinsBidOrderPlacement parseAttachment(JSONObject attachmentData)
...
+        try
+        {
+            if (senderAccount.getUnconfirmedBalanceATM() >= Math.multiplyExact(attachment.getQuantityATU(), attachment.getPriceATM())) {
+                getAccountService().addToUnconfirmedBalanceATM(senderAccount, getLedgerEvent(), transaction.getId(), -Math.multiplyExact(attachment.getQuantityATU(), attachment.getPriceATM()));
+                return true;
+            }
+            return false;
+        }
+        catch (java.lang.ArithmeticException e)
+        {
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/cc/CCBidOrderPlacementTransactionType.java:73]: Add logging here.
put
`@Slf4j`
on top of class,
put logging into catch block:
`log.error("Error: attachment = {}", attachment);`

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/dgs/PurchaseTransactionType.java [hunk 2]
```diff
@@ -79,11 +79,17 @@ public void doStateIndependentValidation(Transaction transaction) throws AplExce
...
-        }
+        try{
+            if (senderAccount.getUnconfirmedBalanceATM() >= Math.multiplyExact((long) attachment.getQuantity(), attachment.getPriceATM())) {
+                getAccountService().addToUnconfirmedBalanceATM(senderAccount, getLedgerEvent(), transaction.getId(), -Math.multiplyExact((long) attachment.getQuantity(), attachment.getPriceATM()));
+                return true;
+            }
         return false;
+        }
+        catch (java.lang.ArithmeticException e)
+        {
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/dgs/PurchaseTransactionType.java:90]: Add logging here.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/ms/MSPublishExchangeOfferTransactiionType.java [hunk 3]
```diff
@@ -97,12 +97,18 @@ public void doStateIndependentValidation(Transaction transaction) throws AplExce
...
+        try {
+            if (senderAccount.getUnconfirmedBalanceATM() >= Math.multiplyExact(attachment.getInitialBuySupply(), attachment.getBuyRateATM()) && accountCurrencyService.getUnconfirmedCurrencyUnits(senderAccount, attachment.getCurrencyId()) >= attachment.getInitialSellSupply()) {
+                getAccountService().addToUnconfirmedBalanceATM(senderAccount, getLedgerEvent(), transaction.getId(), -Math.multiplyExact(attachment.getInitialBuySupply(), attachment.getBuyRateATM()));
+                accountCurrencyService.addToUnconfirmedCurrencyUnits(senderAccount, getLedgerEvent(), transaction.getId(), attachment.getCurrencyId(), -attachment.getInitialSellSupply());
+                return true;
+            }
         return false;
+        }
+        catch (java.lang.ArithmeticException e)
+        {
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/ms/MSPublishExchangeOfferTransactiionType.java:109]: Add logging here.

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/ms/MSReserveIncreaseTransactionType.java [hunk 4]
```diff
@@ -70,16 +70,23 @@ public void doStateIndependentValidation(Transaction transaction) throws AplExce
...
-        if (senderAccount.getUnconfirmedBalanceATM() >= Math.multiplyExact(currency.getReserveSupply(), attachment.getAmountPerUnitATM())) {
+        try{
+            if (senderAccount.getUnconfirmedBalanceATM() >= Math.multiplyExact(currency.getReserveSupply(), attachment.getAmountPerUnitATM())) {
             getAccountService().addToUnconfirmedBalanceATM(senderAccount, getLedgerEvent(), transaction.getId(), -Math.multiplyExact(currency.getReserveSupply(), attachment.getAmountPerUnitATM()));
             return true;
         }
         return false;
+        }
+        catch (java.lang.ArithmeticException e)
+        {
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/ms/MSReserveIncreaseTransactionType.java:86]: Add logging here.
put
`@Slf4j`
on top of class,
put logging into catch block:
`log.error("Error: currency = {}, attachment = {}", currency, attachment);`

**Review by yuriy-larin-firstbridge — APPROVED** [review]

**Review by alzinchenko — APPROVED** [review]

**Pull request merged**